doc/internals/export.rst

```diff
@@ -0,0 +1,98 @@
+.. _export:
+
+Exporting Symbols
+=================
+
+The arbor libraries are compiled with hidden visibility by default. Thus, symbols which are part of
+the public API need to be marked explicitely as visible. Arbor provides a couple of macros to
+annotate functions and classes which are defined in the header file ``export.hpp`` in each library's
+include directory, i.e. ``include/arbor/export.hpp``. These header files are generated at configure
+time based on the build variant, compiler and platform.
+
+Macro Descripiton
+-----------------
+
+.. c:macro:: ARB_LIBNAME_API
+
+    Here ``LIBNAME`` is a placeholder for the library's name: ``ARB_ARBOR_API`` for the main arbor
+    library, ``ARB_ARBORIO_API`` for arborio, etc. This macro is intended to annotate functions,
+    classes and structs which need to be accessible when using the library. Note, that it expands to
+    different values when arbor is being built vs. when arbor is being used by an application. Below
+    we list the places where the macro needs to be added or can be safely omitted (we assume all of
+    the symbols below are part of the public API).
```

Comment:
public = not hidden? Earlier you've used 'visible' as opposite for 'hidden'. Prefer consistency (I see the GCC page does not set a good example either).

Reply:
public in the sense of accessible (vs. private, implementation details). Public/private is an orthogonal concept to hidden/visible. Both public and private symbols are affected by visibility. I could replace "public API" with user-facing API, would that help? I don't think I've used public elsewhere to describe non-hidden - that would indeed be confusing.

Reply:
It's clear now that I am not an expert. My knowledge is limited to private/public/protected properties/methods, which are sometimes described as visibility, and exported symbols (or symbols not exported, which you might describe as hidden).
It seems that hidden/visible has another specific meaning which I struggle to disentangle from what one learns in every first year C++ course.

Is it possible to have a line or two right at the beginning of the page that clarify the difference?

Reply:
done

Reply:
Thanks!